### PR TITLE
Added support for

### PR DESCRIPTION
**Pull request by llyys**
- standard functions 
```
attrs = jsonencode({foo = 'bar'})
```
- negative numbers 
```
y = -10
```

### CODE REVIEW

**Comment by tmccombs** [comment]
The negative number part of this, I think is great. 

The standard functions part, I'm not as sure about. As I understand this, the change is actually evaluating the function during the conversion to json, which can lose some of the original information (like the dependencies on the arguments to the function). It also only works for certain functions, which, for example doesn't work for all functions supported by terraform, and I think would cause this to error instead of translating to hcl-compatible json. 

I'm curious what your use-case is for evaluating functions at translation time? I'd also be more ok with it if it was behind a flag.

**Comment by llyys** [comment]
The problem is that terraform does not support nested maps
https://github.com/hashicorp/terraform-plugin-sdk/issues/62
So we are overcoming this with encoding the payload as json string
```
resource "statechart" "default" {
  name = "Simple chart"
  definition = jsonencode({
     type="start"
     properties={
        label="Start"
        offset={
            x=10
            y=-10
        } 
     }
  })
}
```

**Comment by tmccombs** [comment]
So, currently if you have something like `attrs = jsonencode({foo="foo"})`, then hcl2json will encode that as:

```
"attrs": "${jsonencode({foo=\"foo\"})}"
```

And terraform can handle this syntax just fine. 

It sounds like you want it to be encoded as:

```
"attrs": "{\"foo\":\"foo\"}"
```

I'm still not sure what the use case for that is.

I also don't want to accept this PR as is, because if there is a function hcl2json doesn't know about, or it can't evaluate an argument to a function (for example because it uses a reference to another resource in terraform), then it will fail. I might accept it if hcl2json gracefully falls back to the current behavior if it is unable to  successfully evaluate the function, especially if there is a flag to toggle the behavior.

**Comment by tmccombs** [comment]
I added the negative number support in e4347dbe8d7692530c980f95f432ba10de6db271

File: convert.go [hunk 1]
```diff
@@ -90,10 +111,15 @@ func (c *converter) convertExpression(expr hclsyntax.Expression) (interface{}, e
 	switch value := expr.(type) {
 	case *hclsyntax.LiteralValueExpr:
 		return ctyjson.SimpleJSONValue{Value: value.Val}, nil
+	case *hclsyntax.UnaryOpExpr:
```
Comment (tmccombs) [inline convert.go:114]: This functionality is already on master.

File: convert.go [hunk 2]
```diff
@@ -219,3 +245,14 @@ func (c *converter) convertTemplateFor(expr *hclsyntax.ForExpr) (string, error)
...
 	return "${" + c.rangeSource(expr.Range()) + "}"
 }
+
+func (c *converter) convertFunctionCall(expr hclsyntax.Expression) (interface{}, error) {
+	specCtx := &hcl.EvalContext{
+		Functions: specFuncs,
+	}
+	value, valueDiags := expr.Value(specCtx)
+	if valueDiags.HasErrors() {
+		return nil, fmt.Errorf(valueDiags.Error())
```
Comment (tmccombs) [inline convert.go:255]: This should fall back to the existing behavior of wrapping the expression.

**Review by tmccombs — CHANGES_REQUESTED** [review]

File: convert.go (hunk 2 above)
Comment (llyys) [reply convert.go:255]: Yeah it's not that important.

**Comment by tmccombs** [comment]
https://github.com/tmccombs/hcl2json/commit/b59e3a3b312da99d35d0560919af2619328a482c may interest you.

**Pull request closed without merging**